### PR TITLE
made get sites and get stewardship activities public

### PR DESCRIPTION
**Pull request by willmt80**
Split the site router into a public routes and protected routes. Users will be able to view site information and stewardship activities on tree pages even if they aren't logged in.

### CODE REVIEW

File: service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    this.db = db;
+  }
+
+  private void checkSiteExists(int siteId) {
+    if (!db.fetchExists(db.selectFrom(SITES).where(SITES.ID.eq(siteId)))) {
+      throw new ResourceDoesNotExistException(siteId, "Site");
+    }
+  }
+
+  private Boolean isAlreadyAdopted(int userId, int siteId) {
```
Comment (connernilsen) [inline service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java:40]: Just for clarification, a site can be adopted by multiple users? If that's true, then maybe change the method name here to `isAlreadyAdoptedByUser()` to make it more clear.

File: service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+  }
+
+  @Override
+  public void unadoptSite(JWTData userData, int siteId) {
+    checkSiteExists(siteId);
+    if (!isAlreadyAdopted(userData.getUserId(), siteId)) {
+      throw new WrongAdoptionStatusException(false);
+    }
+
+    db.deleteFrom(ADOPTED_SITES)
```
Comment (connernilsen) [inline service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java:78]: I haven't seen the schema for this db in a while, but just to double check, are we not doing soft deletes on this table?

**Review by connernilsen — COMMENTED** [review]
Good work! I just have a few questions before I approve

File: service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java (hunk 2 above)
Comment (willmt80) [reply service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java:78]: I'm not sure, @florisdobber @oloveluck do either of you have thoughts on this?

File: service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java (hunk 1 above)
Comment (willmt80) [reply service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java]: No, a site can only be adopted by 1 person. Added the isAlreadyAdoptedByUser method for unadopting.

**Review by connernilsen — APPROVED** [review]
I think it looks good 👍

File: service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+            .where(ADOPTED_SITES.USER_ID.eq(userId))
+            .and(ADOPTED_SITES.SITE_ID.eq(siteId)));
+  }
+
+  /**
+   * Throws an exception if the user is not an admin or super admin.
+   *
+   * @param level the privilege level of the user calling the route
+   */
+  void isAdminCheck(PrivilegeLevel level) {
```
Comment (rymaju) [inline service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java:56]: ❓ Should this belong in its own utility file?

File: service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+    return db.fetchExists(
+        db.selectFrom(ADOPTED_SITES)
+            .where(ADOPTED_SITES.USER_ID.eq(userId))
+            .and(ADOPTED_SITES.SITE_ID.eq(siteId)));
+  }
+
+  /**
+   * Throws an exception if the user is not an admin or super admin.
+   *
+   * @param level the privilege level of the user calling the route
```
Comment (rymaju) [inline service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java:54]: ⛏️ Should have a `@throws` javadoc

**Review by rymaju — COMMENTED** [review]
Overall looks good to me 👍

**Review by florisdobber — APPROVED** [review]

**Pull request merged**